### PR TITLE
fix(mysql): TIMESTAMP -> CAST

### PR DESCRIPTION
**Pull request by barakalon**
cc @ginter

### CODE REVIEW

File: sqlglot/dialects/mysql.py [hunk 1]
```diff
@@ -243,6 +243,10 @@ class Parser(parser.Parser):
                 format=exp.Literal.string("%B"),
             ),
             "STR_TO_DATE": _str_to_date,
+            "TIMESTAMP": lambda args: exp.cast(
+                args[0],
```
Comment (barakalon) [inline sqlglot/dialects/mysql.py:247]: note: didn't use `seq_get` because it returns an optional type, and cast doesn't accept an optional

**Review by tobymao — APPROVED** [review]
why is this needed?

File: sqlglot/dialects/mysql.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/dialects/mysql.py:247]: you could avoid the cast then

exp.Cast(this=seq_get(), ...)

**Comment by barakalon** [comment]
trying to transpile `TIMESTAMP(x)` to presto, which doesnt have such a function

File: sqlglot/dialects/mysql.py (hunk 1 above)
Comment (barakalon) [reply sqlglot/dialects/mysql.py]: I changed it up a bit after reading mysql docs

**Comment by tobymao** [comment]
> trying to transpile `TIMESTAMP(x)` to presto, which doesnt have such a function

wouldn't it make more sense to have presto generate the cast?

**Review by tobymao — CHANGES_REQUESTED** [review]
let's do this on generation

**Comment by barakalon** [comment]
switched to generation. I'm a little scared because TIMESTAMP is also a type

**Comment by tobymao** [comment]
bigquery also has timestamp func

**Comment by barakalon** [comment]
Hm, yeah.
https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#timestamp

Seems like current implementation will still work, although we're overloading the `expression` arg for mysql and bigquery.

**Review by tobymao — APPROVED** [review]

**Pull request merged**